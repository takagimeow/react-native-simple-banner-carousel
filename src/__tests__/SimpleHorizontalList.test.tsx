/* eslint-disable @typescript-eslint/unbound-method */
import React from 'react';
import 'react-native';
import { render } from '@testing-library/react-native';

import { Card, SimpleHorizontalList } from '..';

describe('SimpleHorizontalList', () => {
  it('Snapshot Test', () => {
    const { toJSON } = render(
      <SimpleHorizontalList
        data={[
          {
            title: 'マクドナルド札幌すすきの店',
            description: '配送手数料: ￥150・5-15分',
            score: 4.7,
            source: {
              uri: 'https://github.com/takagimeow/react-native-simple-banner-carousel',
            },
          },
          {
            title: 'ケンタッキーフライドチキン すすきのノルベサ店',
            description: '配送手数料: ￥100・20-30分',
            score: 4.7,
            source: {
              uri: 'https://github.com/takagimeow/react-native-simple-banner-carousel',
            },
          },
          {
            title: 'すき家 札幌駅前通北一条店',
            description: '配送手数料: ￥100・5-15分',
            score: 4.7,
            source: {
              uri: 'https://github.com/takagimeow/react-native-simple-banner-carousel',
            },
          },
          {
            title: '大戸屋ごはん処 ニッセイ札幌ビル店',
            description: '配送手数料: ￥100・30-40分',
            score: 4.7,
            source: {
              uri: 'https://github.com/takagimeow/react-native-simple-banner-carousel',
            },
          },
          {
            title: 'スターバックス コーヒー 札幌パルコ',
            description: '配送手数料: ￥100・5-15分',
            score: 4.8,
            source: {
              uri: 'https://github.com/takagimeow/react-native-simple-banner-carousel',
            },
          },
        ]}
        renderItem={(props, i, width) => {
          return (
            <Card
              id={`${props.title}_${i}`}
              width={width}
              {...props}
              onPress={() => console.log('hello world')}
            />
          );
        }}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
