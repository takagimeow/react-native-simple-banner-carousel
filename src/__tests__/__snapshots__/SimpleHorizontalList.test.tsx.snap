// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleHorizontalList Snapshot Test 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    alwaysBounceVertical={false}
    bounces={false}
    collapsable={false}
    contentContainerStyle={
      Array [
        Object {
          "paddingHorizontal": 0,
        },
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    decelerationRate="fast"
    horizontal={true}
    nativeID="animatedComponent"
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="start"
    snapToInterval={578.5}
    style={
      Object {
        "backgroundColor": "transparent",
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fffffff",
                "maxWidth": "100%",
                "paddingBottom": 16,
                "paddingTop": 12,
                "width": 562.5,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://github.com/takagimeow/react-native-simple-banner-carousel",
              }
            }
            style={
              Object {
                "backgroundColor": "#bbb",
                "height": 156,
                "marginBottom": 8,
                "maxWidth": "100%",
                "width": 562.5,
              }
            }
          />
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 11,
                  "fontSize": 14,
                }
              }
            >
              マクドナルド札幌すすきの店
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f0f0f0",
                  "borderRadius": 25,
                  "display": "flex",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 4,
                  "paddingVertical": 6,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                4.7
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              配送手数料: ￥150・5-15分
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fffffff",
                "maxWidth": "100%",
                "paddingBottom": 16,
                "paddingTop": 12,
                "width": 562.5,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://github.com/takagimeow/react-native-simple-banner-carousel",
              }
            }
            style={
              Object {
                "backgroundColor": "#bbb",
                "height": 156,
                "marginBottom": 8,
                "maxWidth": "100%",
                "width": 562.5,
              }
            }
          />
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 11,
                  "fontSize": 14,
                }
              }
            >
              ケンタッキーフライドチキン すすきのノルベサ店
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f0f0f0",
                  "borderRadius": 25,
                  "display": "flex",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 4,
                  "paddingVertical": 6,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                4.7
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              配送手数料: ￥100・20-30分
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fffffff",
                "maxWidth": "100%",
                "paddingBottom": 16,
                "paddingTop": 12,
                "width": 562.5,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://github.com/takagimeow/react-native-simple-banner-carousel",
              }
            }
            style={
              Object {
                "backgroundColor": "#bbb",
                "height": 156,
                "marginBottom": 8,
                "maxWidth": "100%",
                "width": 562.5,
              }
            }
          />
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 11,
                  "fontSize": 14,
                }
              }
            >
              すき家 札幌駅前通北一条店
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f0f0f0",
                  "borderRadius": 25,
                  "display": "flex",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 4,
                  "paddingVertical": 6,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                4.7
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              配送手数料: ￥100・5-15分
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fffffff",
                "maxWidth": "100%",
                "paddingBottom": 16,
                "paddingTop": 12,
                "width": 562.5,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://github.com/takagimeow/react-native-simple-banner-carousel",
              }
            }
            style={
              Object {
                "backgroundColor": "#bbb",
                "height": 156,
                "marginBottom": 8,
                "maxWidth": "100%",
                "width": 562.5,
              }
            }
          />
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 11,
                  "fontSize": 14,
                }
              }
            >
              大戸屋ごはん処 ニッセイ札幌ビル店
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f0f0f0",
                  "borderRadius": 25,
                  "display": "flex",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 4,
                  "paddingVertical": 6,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                4.7
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              配送手数料: ￥100・30-40分
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fffffff",
                "maxWidth": "100%",
                "paddingBottom": 16,
                "paddingTop": 12,
                "width": 562.5,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://github.com/takagimeow/react-native-simple-banner-carousel",
              }
            }
            style={
              Object {
                "backgroundColor": "#bbb",
                "height": 156,
                "marginBottom": 8,
                "maxWidth": "100%",
                "width": 562.5,
              }
            }
          />
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 11,
                  "fontSize": 14,
                }
              }
            >
              スターバックス コーヒー 札幌パルコ
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#f0f0f0",
                  "borderRadius": 25,
                  "display": "flex",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 4,
                  "paddingVertical": 6,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                4.8
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              配送手数料: ￥100・5-15分
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
